gcsfs/dask_link.py

```diff
@@ -118,7 +120,6 @@ def walk(self, path, refresh=False):
     dask.bytes.core._filesystems['gs'] = DaskGCSFileSystem
 
 
-try:
+# Register to dask if dask is installed
+if 'dask' in sys.modules:
```

Comment:
This doesn't do what you think, it requires dask to have been already imported. Therefore, if gcsfs happens to be imported before dask, register will never be called.